examples/golang/README.md

```diff
@@ -0,0 +1,41 @@
+# Golang Examples
+
+This folder contains examples that can be run with [wasmcloud][wasmcloud] which are written in [Golang][golang].
+
+Most examples compile with [TinyGo][tinygo] rather than the Golang compiler proper given that WebAssembly support landed in TinyGo first and is still slightly better supported there.
+
+## WebAssembly support
+
+As WebAssembly is intended to be a "compilation target" for native Golang code, upstream work is underway to integrate and improve support for the various standards of WebAssembly.
+
+| Language | Core Modules (`wasm32-unknown-unknown`) | Preview 1 (`wasm32-wasi-preview1`) | WASI Preview 2 (`wasm32-wasi-preview2`) |
+|----------|-----------------------------------------|------------------------------------|-----------------------------------------|
+| Golang   | ✅ (`GOOS=js`,`GOARCH=wasm`)            | ✅ (`GOOS=wasip1`)                 | 🏗                                       |
+| TinyGo   | ✅ (`-target=wasm`)                     | ✅ (`-target=wasi`)                | 🏗                                       |
```

Comment:
I think these tables are technically correct (the best kind of correct?), but will likely create the wrong impression for new users. I.e. while TinyGo, Rust, and JS/TS toolchains currently target WASI P1 from a _compiler_ perspective, the tooling (e.g. jco, cargo component, wash) will automatically adapt to a WASI P2 component, ready to be run in wasmtime, wasmCloud, etc.

So from a user perspective, I would expect to see a ✅ for some of these P2 columns

Reply:
Yeah I was trying to focus here on the baked-into-the-language functionality -- I'll add a note like "requires adapter"

Reply:
done, updated!